### DISCUSSION TITLE
bulk collection updates without loading the entities

### DISCUSSION THREAD

**Discussion by BrannJoly** [mikro-orm/mikro-orm#5458] [Q&A]
I know I can update some properties without loading the entities as described in the docs [here](https://mikro-orm.io/docs/entity-manager#updating-references-not-loaded-entities).

I was wondering if something similar was possible with collections.

Using the [books and tags ](https://mikro-orm.io/docs/guide/relationships#collections-onetomany-and-manytomany)example of the official docs, what if I want to add a tag to 10.000 books in one batch?
is there a way to do this without having to first load the 10.000 books? 
Right now I'm using a stored procedure, but it would be amazing if it could be managed somehow by mikro-orm!

Thank you

**Comment by B4nan** [comment]
I believe you should be able to do the same, just create the reference via `em.create()` so it has the collection instances present - it doesn't have to be initialized to add a value, and the queries will be batched automatically even for the collection updates.

```ts
const book = em.create(Book, { id: 123 }, { managed: true, partial: true });
book.tags.add(new BookTag(...));
await em.flush();
```
